src/main/java/com/aws/greengrass/util/S3SdkClientFactory.java

```diff
@@ -34,16 +34,18 @@
      * @param credentialsProvider credential provider from TES
      */
     @Inject
-    @SuppressWarnings("PMD.AvoidCatchingGenericException")
+    @SuppressWarnings({"PMD.AvoidCatchingGenericException", "PMD.NullAssignment"})
     public S3SdkClientFactory(DeviceConfiguration deviceConfiguration, LazyCredentialProvider credentialsProvider) {
         this.credentialsProvider = credentialsProvider;
-        try {
-            deviceConfiguration.validate(true);
-        } catch (DeviceConfigurationException e) {
-            configValidationError = e.getMessage();
-        }
-        deviceConfiguration.getAWSRegion().subscribe((what, node) ->
-                this.s3Client = getClientForRegion(Region.of(Coerce.toString(deviceConfiguration.getAWSRegion()))));
+        deviceConfiguration.getAWSRegion().subscribe((what, node) -> {
+            this.s3Client = getClientForRegion(Region.of(Coerce.toString(deviceConfiguration.getAWSRegion())));
+            try {
+                deviceConfiguration.validate();
+                configValidationError = null;
+            } catch (DeviceConfigurationException e) {
+                configValidationError = e.getMessage();
+            }
```

Comment:
move this before doing the region thing

Reply:
It does not matter, we are not preventing client creation if validation fails. At least thats how it is right now.

Reply:
It somewhat matters because the line above this check may throw an exception and thus the error message won't be updated with what the proper error is.